src/System.Security.Cryptography.Encoding/src/System/Security/Cryptography/Base64Transforms.cs

```diff
@@ -19,67 +18,54 @@ public enum FromBase64TransformMode
     {
         // converting to Base64 takes 3 bytes input and generates 4 bytes output
-        public int InputBlockSize
-        {
-            get { return (3); }
-        }
+        public int InputBlockSize => 3;
+        public int OutputBlockSize => 4;
+        public bool CanTransformMultipleBlocks => false;
+        public virtual bool CanReuseTransform => true;
 
-        public int OutputBlockSize
+        public int TransformBlock(byte[] inputBuffer, int inputOffset, int inputCount, byte[] outputBuffer, int outputOffset)
         {
-            get { return (4); }
-        }
+            ValidateTransformBlock(inputBuffer, inputOffset, inputCount);
 
-        public bool CanTransformMultipleBlocks
-        {
-            get { return (false); }
+            // For now, only convert 3 bytes to 4
+            byte[] tempBytes = ConvertToBase64(inputBuffer, inputOffset, 3);
+
+            Buffer.BlockCopy(tempBytes, 0, outputBuffer, outputOffset, tempBytes.Length);
+            return tempBytes.Length;
         }
 
-        public virtual bool CanReuseTransform
+        public byte[] TransformFinalBlock(byte[] inputBuffer, int inputOffset, int inputCount)
         {
-            get { return (true); }
+            ValidateTransformBlock(inputBuffer, inputOffset, inputCount);
+
+            // Convert.ToBase64CharArray already does padding, so all we have to check is that
+            // the inputCount wasn't 0
+            if (inputCount == 0)
+            {
+                return Array.Empty<byte>();
+            }
+
+            // Again, for now only a block at a time
+            return ConvertToBase64(inputBuffer, inputOffset, inputCount);
         }
 
-        public int TransformBlock(byte[] inputBuffer, int inputOffset, int inputCount, byte[] outputBuffer, int outputOffset)
+        private byte[] ConvertToBase64(byte[] inputBuffer, int inputOffset, int inputCount)
         {
-            if (inputBuffer == null) throw new ArgumentNullException(nameof(inputBuffer));
-            if (inputOffset < 0) throw new ArgumentOutOfRangeException(nameof(inputOffset), SR.ArgumentOutOfRange_NeedNonNegNum);
-            if (inputCount < 0 || (inputCount > inputBuffer.Length)) throw new ArgumentException(SR.Argument_InvalidValue);
-            if ((inputBuffer.Length - inputCount) < inputOffset) throw new ArgumentException(SR.Argument_InvalidOffLen);
-
-            // For now, only convert 3 bytes to 4
             char[] temp = new char[4];
-            Convert.ToBase64CharArray(inputBuffer, inputOffset, 3, temp, 0);
-
+            Convert.ToBase64CharArray(inputBuffer, inputOffset, inputCount, temp, 0);
             byte[] tempBytes = Encoding.ASCII.GetBytes(temp);
-            if (tempBytes.Length != 4) throw new CryptographicException(SR.Cryptography_SSE_InvalidDataSize);
+            if (tempBytes.Length != 4)
+                throw new CryptographicException(SR.Cryptography_SSE_InvalidDataSize);
```

Comment:
I just added additional tests. The To case doesn't work (due to hard-coded 4 byte buffer) but the From case does. E.g. this fails (in desktop and corefx)

            byte[] data_5bytes = System.Text.Encoding.ASCII.GetBytes("aaaaa");
            using (var transform = new ToBase64Transform())
                transform.TransformFinalBlock(data_5bytes, 0, 5);